homeassistant/components/ozw/light.py

```diff
@@ -42,23 +60,80 @@ def byte_to_zwave_brightness(value):
     return 0
 
 
-class ZwaveDimmer(ZWaveDeviceEntity, LightEntity):
-    """Representation of a Z-Wave dimmer."""
+# Create ZwaveLight class and combine them
+class ZwaveLight(ZWaveDeviceEntity, LightEntity):
+    """Representation of a Z-Wave light."""
 
     def __init__(self, values):
         """Initialize the light."""
         super().__init__(values)
+        self._color_channels = None
+        self._hs = None
+        self._ct = None
+        self._white = None
         self._supported_features = SUPPORT_BRIGHTNESS
         # make sure that supported features is correctly set
         self.on_value_update()
 
     @callback
     def on_value_update(self):
         """Call when the underlying value(s) is added or updated."""
-        self._supported_features = SUPPORT_BRIGHTNESS
         if self.values.dimming_duration is not None:
             self._supported_features |= SUPPORT_TRANSITION
 
+        if self.values.color is not None:
+            self._supported_features |= SUPPORT_COLOR
```

Comment:
This is something we need to test properly and the reason why both I and @MartinHjelmare did not yet copy the rgb support from the old implementation into this new one. Values come in in a (semi) random order. So it might happen that a light entity is created with only white/dimmer support as the rgb values were not yet received. In that case connected apps/appliances such as the Google Home might discover the light as dimmable light only, even if the supported features get extended later with the color support.

Maybe we should just continue with this approach and see where it goes @MartinHjelmare and @cgarwood ?
If we get some reports about RGB Zwave lights in relation to Google Home/Alexa we can see from there.

One way to work around it is to lookup the values in the tree, as the upstream library has some logic to hold all values belonging to a CC until all are received. So for example for self.values.color_channels you are relying on the fact that you already received that value and it has been added to the values object. If this is causing trouble we can do something like this:

```
color_channels = None
for item in self.values.primary.node.values():
    if item.command_class == CommandClass.THECOMMANDCLASSWENEED:
        for value in in item.values():
           if value.index == THEINDEXFORCOLOR:
                  color_channels = value.value
```

We can also create a helper method in the lib like "has_value(cc_id, val_id)" or something like that.
This is just from the top of my head but I'm actually sure this works. But like I said before: maybe we should go for the easy route at first and this issue will never even happen ?

Reply:
I'll test this branch next week. In my previous attempt the Aeotec light I tested with failed to present in the supported order.

Reply:
OK, if you want I can go ahead and create a helper method in the lib anyway so we can do these kind of checks

Reply:
A helper might be good.

But the correct presentation might still be delayed, right? Does the upstream library really know all present CCs before they are reported by the daemon?

Reply:
@balloob might be able to tell more about that but basically that is indeed what's happening. It's queueing all messages from one topic until it received all values in that commandclass-topic. 
https://github.com/cgarwood/python-openzwave-mqtt/blob/master/openzwavemqtt/base.py

So it waits until all children of a topic are received before dequeueing it.
From what I have seen this also applies to the entire node object.

Reply:
Ok, so when we receive the first message from the library for one value we know that all the values for a node should be populated and we can query the library node value tree.

Reply:
But how does the library know what *all* values is for a node? Does the daemon send some ready event?

Reply:
> Ok, so when we receive the first message from the library for one value we know that all the values for a node should be populated and we can query the library node value tree.

Correct, that's how it should work and quickly reviewing the code it indeed looks like it but @balloob wrote most part of that lib so he might have some additional comments.

Reply:
> But how does the library know what _all_ values is for a node? Does the daemon send some ready event?

When another topic is received, it dequeues the items from the previous topic. This all still implies that OZW is actually sending them in order btw.

Reply:
What does this really mean?

> To prevent signals being sent out of order, we will hold all messages for children until the parent has received its information.

Reply:
Ok, I think I get it now.

Reply:
It's queueing up all info until another item in the tree is received and move up.
queue all items under a node until we receive info from another node. If that happens, dequeue all info from teh previous node.
Again, this still implies that OZW is sending the info serialized node-by-node but from what I see it's actually doing that.

Reply:
The only time this fails btw is when adding a NEW node to the mesh. In that case info comes in in chunks. We learned that the hard way with the naming stuff a few months back.

Reply:
So let's go for the value lookup approach on first value then. :+1:

Reply:
> I'll test this branch next week. In my previous attempt the Aeotec light I tested with failed to present in the supported order.

A few users with RGB Lights have tested this with their Inovelli LW42 bulbs and report having RGB color controls. So if anything I'd say it's a great starting point.

Reply:
Also `self.values.color` and `self.value.color_channels` are coming from the CC via discovery.

Reply:
Do you know if any of these users have a Google Home / Alexa attached ? So they can check if the rgb color controls are present on the smart speaker ? What we saw previously was the values coming in serialized so for example the light gets created without the color feature (as that value was not yet received by the EntityValues object) and a few (milli)seconds later the color feature is added when the color value comes in but the entity without color support is already published to externally connected apps/appliances. So, what I mean is the features of a device change and that caused some issues. 

If that's still an issue we can implement an helper in the library to get check for those features without relying on the values object. That's where I was talking about with @MartinHjelmare. 

I think I will have some time later today to implement the helper for you in the lib and then I'll help you out where to use it in this PR.

Reply:
So about the supported features. We sync Google every restart. So if the supported features are cached between restarts and we have the right supported features at next boot, it would be ok in Google at least after the next restart and onwards.

Reply:
We've decided to go ahead with this PR without major changes and use it as a base to build on.